File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.java
class Solution {
    public String solution(String X, String Y) {
        StringBuffer answer = new StringBuffer("");
        int[] x = {0,0,0,0,0,0,0,0,0,0};
        int[] y = {0,0,0,0,0,0,0,0,0,0};
        boolean[] key = {true, true};
        
        for(int i=0; i<X.length() || i<Y.length(); i++){
            if(i<X.length())
                x[X.charAt(i)-'0']++;
            if(i<Y.length())
                y[Y.charAt(i)-'0']++;
        }
        
        for(int i=9; i>=0; i--){
            if(x[i]!=0 && y[i]!=0){
                key[0] = false;
                for(int j=0; j<x[i] && j<y[i]; j++){
                    answer.append(i);
                }
                if(i!=0) key[1] = false;
            }
        }
        
        if(key[0]){
            answer.append(-1);
        }else if(key[1]){
            answer = new StringBuffer("0");
        }
        
        return answer.toString();
    }
}

File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    var answer = '';
    var x = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
    var y = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
    [...X].forEach(n => x[n]++);
    [...Y].forEach(n => y[n]++);
    var xy = x.map((n, idx) => Math.min(n, y[idx]));
    
    for(var i=9; i>=0; i--) {
        answer += (i+'').repeat(xy[i])
    }
    
    if(!answer) return "-1"
    if(!answer.replace(/0/g, "")) return "0"
    return answer;
}